packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,688 @@
+    const haveName = Boolean(attrs.launchTemplateName);
+    if (haveId == haveName) {
+      throw new Error('LaunchTemplate.fromLaunchTemplateAttributes() requires exactly one of launchTemplateId or launchTemplateName be provided.');
+    }
+
+    class Import extends Resource implements ILaunchTemplate {
+      public readonly versionNumber = attrs.versionNumber ?? LaunchTemplateSpecialVersions.DEFAULT_VERSION;
+      public readonly launchTemplateId? = attrs.launchTemplateId;
+      public readonly launchTemplateName? = attrs.launchTemplateName;
+    }
+    return new Import(scope, id);
+  }
+
+  // ============================================
+  //   Members for ILaunchTemplate interface
+  /**
+   * @inheritdoc
+   */
+  public readonly versionNumber: string;
+
+  /**
+   * @inheritdoc
+   */
+  public readonly launchTemplateId?: string;
+
+  /**
+   * @inheritdoc
+   */
+  public readonly launchTemplateName?: string;
+
+  // =============================================
+  //   Data members
+
+  /**
+   * The default version for the launch template.
+   *
+   * @attribute
+   */
+  public readonly defaultVersionNumber: string;
+
+  /**
+   * The latest version of the launch template.
+   *
+   * @attribute
+   */
+  public readonly latestVersionNumber: string;
+
+  /**
+   * IAM Role assumed by instances that are launched from this template.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * Principal to grant permissions to.
+   */
+  public readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * Allows specifying security group connections for the instance.
+   *
+   * Note: If the LaunchTemplate was not created with a SecurityGroup, then this connections
+   * object will be empty and operations on it will do nothing.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * UserData executed by instances that are launched from this template.
+   */
+  public readonly userData?: UserData;
+
+
+  protected tagSpecifications: LaunchTemplateTagSpecification[];
+
+  // =============================================
+
+  constructor(scope: Construct, id: string, props?: LaunchTemplateProps) {
+    super(scope, id);
+
+    // Basic validation of the provided spot block duration
+    const spotDuration = props?.spotOptions?.blockDuration?.toHours({ integral: true });
+    if (spotDuration !== undefined && (spotDuration < 1 || spotDuration > 6)) {
+      // See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances
+      Annotations.of(this).addError('Spot block duration must be exactly 1, 2, 3, 4, 5, or 6 hours.');
+    }
+
+    this.role = props?.role ?? new iam.Role(this, 'Role', {
+      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+    });
+    this.grantPrincipal = this.role;
+    const iamProfile = new iam.CfnInstanceProfile(this, 'Profile', {
+      roles: [this.role.roleName],
+    });
+
+    if (props?.securityGroup) {
```

Comment:
This seems to be the only property of a LaunchTemplate that could tie it to a specific VPC. So, though it might be nice to be able to create one by default if we're not provided one, the lack of a VPC property on the LaunchTemplate prevents that.

The `IConnectable` interface is really useful, so it seems sensible to me to want to have the L2 LaunchTemplate implement the interface. However, the inability to create a default security group makes the case of not being provided a security group a little weird -- you end up with a Connections object that has no actual security group(s) in it.

Reply:
I see how it will be useful, but I'm still a little hesitant to create one automatically--especially as I can see use cases for Launch Templates where people don't want those (they will use LTs to set Spot Pricing for all instances but then create SGs for individual instances... or something).

I guess we can't!